### PR TITLE
feat: Idp 1918 sviluppo api creazione trx

### PR DESCRIPTION
**Pull request by Giuseppe-LaManna**
#### List of Changes

- Added common creation management

#### How Has This Been Tested?

- Pre-Deploy Test
    - [X] Unit
    - [ ] Integration (Narrow)
- Post-Deploy Test
    - [ ] Isolated Microservice
    - [ ] Broader Integration
    - [ ] Acceptance
    - [ ] Performance & Load

#### Types of changes

- [ ] PATCH - Bug fix (backwards compatible bug fixes)
- [x] MINOR - New feature (add functionality in a backwards compatible manner)
- [ ] MAJOR - Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/payment/controller/payment/CommonPaymentController.java [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-import it.gov.pagopa.payment.dto.qrcode.TransactionResponse;
 import jakarta.validation.Valid;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.*;
-
-@RequestMapping("/idpay/payment/")
+@RequestMapping("/idpay/payment")
 public interface CommonPaymentController {
-    @PostMapping("/merchant")
+    @PostMapping("/create-trx")
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payment/controller/payment/CommonPaymentController.java:10]: review path with @frtrinca 
I think we should let "merchant" somewhere inside the path

File: src/test/java/it/gov/pagopa/payment/controller/payment/CommonPaymentControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private static final String ACQUIRER_ID = "ACQUIRER_ID";
+    private static final String ID_TRX_ISSUER = "ID_TRX_ISSUER";
+    private static final String MERCHANT_ID = "MERCHANT_ID";
+
+//    @Test
```
Comment (antonioT90) [inline src/test/java/it/gov/pagopa/payment/controller/payment/CommonPaymentControllerTest.java:37]: why this commented test?

File: src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionInProgress2BaseTransactionResponseMapper.java
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionInProgress2BaseTransactionResponseMapper.java:1]: add a test for this mapper, checking notNull fields in the result

File: src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionInProgress2BaseTransactionResponseMapper.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+@Service
+@Slf4j
+public class TransactionInProgress2BaseTransactionResponseMapper
+    implements Function<TransactionInProgress, BaseTransactionResponseDTO> {
+
+  private final int authorizationExpirationMinutes;
+
+  public TransactionInProgress2BaseTransactionResponseMapper(@Value("${app.qrCode.expirations.authorizationMinutes}") int authorizationExpirationMinutes) {
+    this.authorizationExpirationMinutes = authorizationExpirationMinutes; //TODO OP
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionInProgress2BaseTransactionResponseMapper.java:19]: resolve this TODO

File: src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionInProgress2BaseTransactionResponseMapper.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            .merchantId(transactionInProgress.getMerchantId())
+            .trxDate(transactionInProgress.getTrxDate())
+            .trxCode(transactionInProgress.getTrxCode())
+            .status(transactionInProgress.getStatus())
+            .merchantFiscalCode(transactionInProgress.getMerchantFiscalCode())
+            .vat(transactionInProgress.getVat())
+            .splitPayment(splitPayment)
+            .residualAmountCents(residualAmountCents)
+            .trxExpirationMinutes(authorizationExpirationMinutes)
+            .build();
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionInProgress2BaseTransactionResponseMapper.java:48]: you are not filling txt and png url fields

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/test/java/it/gov/pagopa/payment/controller/payment/CommonPaymentControllerTest.java (hunk 2 above)
Comment (Giuseppe-LaManna) [reply src/test/java/it/gov/pagopa/payment/controller/payment/CommonPaymentControllerTest.java:37]: I'm still developing this test

**Pull request merged**